src/interfaces/statics.rs

```diff
@@ -401,139 +410,262 @@ mod tests {
+            };
+
+            let mut de = Deserializer::new(ciphertext.as_slice());
+            let encrypted_header = EncryptedHeader::read(&mut de)?;
+            let ciphertext = de.finalize();
+            let cover_crypt = CoverCryptX25519Aes256::default();
+
+            let plaintext_header =
+                encrypted_header.decrypt(&cover_crypt, &user_key, authentication_data)?;
+            assert_eq!(plaintext_header.additional_data, header_metadata);
+            let plaintext = cover_crypt.decrypt(
+                &plaintext_header.symmetric_key,
+                &ciphertext,
+                authentication_data,
+            )?;
+            assert_eq!(expected_plaintext, plaintext);
+
+            Ok(())
+        }
 
-        //
-        // Uncomment the following code to write a new regression vector
-        //
+        pub fn new(
+            mpk: &PublicKey,
+            policy: &Policy,
+            encryption_policy: &str,
+            plaintext: &str,
+            header_metadata: Option<&[u8]>,
+            authentication_data: Option<&[u8]>,
+        ) -> Result<EncryptionTestVector, Error> {
+            let cover_crypt = CoverCryptX25519Aes256::default();
+            let (symmetric_key, encrypted_header) = EncryptedHeader::generate(
+                &cover_crypt,
+                policy,
+                mpk,
+                &AccessPolicy::from_boolean_expression(encryption_policy)?,
+                header_metadata,
+                authentication_data,
+            )?;
+
+            let mut aes_ciphertext =
+                cover_crypt.encrypt(&symmetric_key, plaintext.as_bytes(), authentication_data)?;
+            let mut ciphertext = encrypted_header.try_to_bytes()?;
+            ciphertext.append(&mut aes_ciphertext);
+            let header_metadata = match header_metadata {
+                Some(ad) => base64::encode(ad),
+                None => String::new(),
+            };
+            let authentication_data = match authentication_data {
+                Some(ad) => base64::encode(ad),
+                None => String::new(),
+            };
+            Ok(EncryptionTestVector {
+                encryption_policy: encryption_policy.to_string(),
+                plaintext: base64::encode(plaintext),
+                ciphertext: base64::encode(ciphertext),
+                header_metadata,
+                authentication_data,
+            })
+        }
+    }
 
-        //let mut encrypted_bytes = encrypted_header.try_to_bytes()?;
-        //encrypted_bytes.append(&mut ctx);
+    #[derive(Debug, Serialize, Deserialize)]
+    struct UserSecretKeyTestVector {
+        access_policy: String,
+        key: String,
+    }
+    impl UserSecretKeyTestVector {
+        pub fn new(
+            msk: &MasterSecretKey,
+            policy: &Policy,
+            access_policy: &str,
+        ) -> Result<Self, Error> {
+            Ok(Self {
+                key: base64::encode(
```

Comment:
Why do we need to encode in base64 everywhere ? The size in byte is unchanged, isn't it ?

Reply:
I tried to find a JSON format than can be easily read by every language (and started with Java where base64 is the default encoding)